docs/oapserver-configuration.md

```diff
@@ -0,0 +1,284 @@
+    - name: SW_CLUSTER_K8S_LABEL
+      value: app=collector,release=skywalking
+    - name: SW_TELEMETRY
+      value: prometheus
+    - name: SW_HEALTH_CHECKER
+      value: default
+    - name: SKYWALKING_COLLECTOR_UID
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.uid
+    - name: SW_LOG_LAL_FILES
+      value: test1
+    - name: SW_LOG_MAL_FILES
+      value: test2
+  # The file configuration of OAPServer
+  # we should avoid setting the same file name in the file
+  file:
+    - name: test1.yaml
+      path: /skywalking/config/lal
+      data: |
+        rules:
+          - name: example
+            dsl: |
+              filter {
+                text {
+                  abortOnFailure false // for test purpose, we want to persist all logs
+                  regexp $/(?s)(?<timestamp>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3}) \[TID:(?<tid>.+?)] \[(?<thread>.+?)] (?<level>\w{4,}) (?<logger>.{1,36}) (?<msg>.+)/$
+                }
+                extractor {
+                  metrics {
+                    timestamp log.timestamp as Long
+                    labels level: parsed.level, service: log.service, instance: log.serviceInstance
+                    name "log_count"
+                    value 1
+                  }
+                }
+                sink {
+                }
+              }
+    - name: test2.yaml
+      path: /skywalking/config/log-mal-rules
+      data: |
+        expSuffix: instance(['service'], ['instance'], Layer.GENERAL)
+        metricPrefix: log
+        metricsRules:
+          - name: count_info
+            exp: log_count.tagEqual('level', 'INFO').sum(['service', 'instance']).downsampling(SUM)
+
+```
+
+
+
+## Spec of OAPServerDynamicConfig
+
+| Field Name    | Description                                            |
+| ------------- | ------------------------------------------------------ |
+| Version       | The version of the OAP server, the default value is 9.0.0      |
+| AllInOne      | All configurations in one CR, the default value is `false` |
+| LabelSelector | The label selector of the specific configmap, the default value is "app=collector,release=skywalking"               |
+| Data          | All configurations' key and value                      |
+
+
+
+## Status of OAPServerDynamicConfig
+
+| Field Name     | Description                                                |
+| -------------- | ---------------------------------------------------------- |
+| State          | The state of dynamic configuration, `running` or `stopped` |
+| CreationTime   | All configurations in one CR, the default value is `false`     |
+| LastUpdateTime | The last time this condition was updated                   |
+
+
+
+## Usage of OAPServerDynamicConfig
+
+There are two ways to configure the OAPServerDynamicConfig. 
+
+- **Global configuration**. Put all the dynamic configurations in one CR, you need to set the field `allInOne` to true.
```

Comment:
Can we provide only one mode here? It's not global or single. we could split all configurations into several  CRs. The controller could check CRs don't overlap each other.

Reply:
Do you mean when users create an all-in-one CR, we need to split the CR's data into several CRs that only contain one configuration?

Reply:
make sense